lib/queue.js

```diff
@@ -1116,7 +1121,12 @@ Queue.prototype.processJob = function(job, notFetch = false) {
     let jobPromise = handler(job);
 
     if (timeoutMs) {
+      let cancelHandler
```

Comment:
Could you explain why this change solves the issue?

Reply:
`pTimeout`, while it cancels the promise that it wraps, doesn't support canceling the resulting promise, and doesn't give any access to the wrapped promise once created.

So, _before_ we wrap the promise, if there is a `cancel` handler on the promise returned from the job handler, this will retain a reference to it that is bound to the context of the original promise. I then attach that method to the promise that `pTimeout()` returns as a cancel handler, so that the end-user can call `cancel()` on the pTimeout-wrapped promise and it will behave exactly as if they called the cancel method on the original promise.

As I write this, I realize that this could (and probably will) result in the `cancel()` method getting called twice, since `pTimeout()` will eventually timeout and call it again.

I see a couple ways to resolve that: we could use pTimeout's `options.customTimers` to pass in a wrapper for setTimeout() that lets us cancel the timer in `cancelHandler()`, or have `cancelHandler` instead be a function that deletes itself from the pTimeout promise when it is called, so that `pTimeout` can't call it a second time. The first seems cleaner, I think.

Reply:
ok, so same comment as above, find a good descriptive name and refactor out to function so that the intent is clear and the code easier to follow.